test/unit/gateways/card_stream_test.rb

```diff
@@ -311,6 +322,47 @@ def test_default_3dsecure_required
     end.respond_with(successful_purchase_response)
   end
 
+  def test_3ds2_data
+    stub_comms do
+      @gateway.purchase(142, @visacreditcard, @visacredit_options.merge(@visacredit_three_ds_options))
+    end.check_request do |_endpoint, data, _headers|
+      assert_match(/threeDSRequired=Y/, data)
+      assert_match(/threeDSEnrolled=Y/, data)
+      assert_match(/threeDSAuthenticated=Y/, data)
+      assert_match(/threeDSECI=5/, data)
+      assert_match(/threeDSCAVV=Y2FyZGluYWxjb21tZXJjZWF1dGg/, data)
+      assert_match(/threeDSXID=00000000000004717472/, data)
+    end.respond_with(successful_purchase_response_with_3dsecure)
+  end
+
+  def test_3ds2_not_enrolled
+    stub_comms do
+      @gateway.purchase(142, @visacreditcard, @visacredit_options.merge(@visacredit_three_ds_options.merge(
```

Comment:
💬  Comment:

Rubocop is not complaining but maybe is a good idea to move that @visacredit_three_ds_options merge one line above just for readability